swift/ql/lib/codeql/swift/frameworks/Alamofire/Alamofire.qll

```diff
@@ -0,0 +1,39 @@
+/**
+ * Models for the Alamofire networking library.
+ */
+
+import swift
+private import codeql.swift.dataflow.DataFlow
+private import codeql.swift.dataflow.ExternalFlow
+private import codeql.swift.dataflow.FlowSources
+
+/**
+ * An Alamofire response handler type.
+ */
+private class AlamofireResponseType extends NominalTypeDecl {
+  AlamofireResponseType() {
+    this.getFullName() = ["DataResponse", "DownloadResponse"] or
+    this.getABaseTypeDecl() instanceof AlamofireResponseType
+  }
+
+  /**
+   * Gets a field that contains remote data.
+   */
+  FieldDecl getADataField() {
+    result = this.getAMember() and
+    result.getName() = ["data", "value", "result"]
+  }
+}
+
+/**
+ * A remote flow source that is an access to remote data from an Alamofire response handler.
+ */
+private class AlamofireResponseSource extends RemoteFlowSource {
+  AlamofireResponseSource() {
+    exists(AlamofireResponseType responseType |
+      this.asExpr().(MemberRefExpr).getMember() = responseType.getADataField()
+    )
+  }
```

Comment:
Could this not be done with the following MaD row:
```qll
";" + ["DataResponse","DownloadResponse"] + ";true;" + ["data", "value", "result"] + ";;;;remote"
```
?

Reply:
I'll try it.  I haven't tried MaD on member variables but I'm guessing it should work...

Reply:
We have a couple of cases of that already in Swift if you need to lookup the syntax

Reply:
Update following a bit more investigation today - for the line:
```
struct DataResponse<Success, Failure: Error> { ...
```
the `Decl` for `DataResponse` is a `StructDecl` and thus a `NominalTypeDecl`.  On the flip side, the corresponding `Type` I get for the `DataResponse` is an `UnboundGenericType` and not a `NominalType`.  This is probably reasonable (I think), but is currently tripping up `interpretElement0`.

As I said before I can fix this, but I'm currently working on the *right* fix that is clean and well understood.  Then I'll make some tests to prove we're really doing the right thing.

Reply:
I decided to put the fix in a separate PR - https://github.com/github/codeql/pull/11514 - as it probably deserves separate review and discussion.  This PR is now blocked by that one.

Reply:
Done.